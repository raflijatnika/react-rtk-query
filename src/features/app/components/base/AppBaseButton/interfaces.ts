// Antd
import { ButtonProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseButtonProps extends ButtonProps {
  backgroundColor?: APP_COLOR | APP_COLOR_LIGHT
  fontColor?: APP_COLOR | APP_COLOR_LIGHT
  width?: number
  height?: number
}
