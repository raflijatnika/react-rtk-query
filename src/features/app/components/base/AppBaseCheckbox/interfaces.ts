// Antd
import { CheckboxProps } from 'antd'

export type IAppBaseCheckboxProps = CheckboxProps
