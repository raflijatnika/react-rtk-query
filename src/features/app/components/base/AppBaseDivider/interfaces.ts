// Antd
import { DividerProps } from 'antd'

export type IAppBaseDividerProps = DividerProps
