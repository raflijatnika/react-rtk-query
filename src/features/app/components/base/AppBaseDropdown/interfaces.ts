// React
import { ReactElement } from 'react'

// Interfaces
import { DropdownProps, MenuProps } from 'antd'

export interface IDropdownProps {
  children?: ReactElement
  trigger?: DropdownProps['trigger']
  items: MenuProps['items']
}
