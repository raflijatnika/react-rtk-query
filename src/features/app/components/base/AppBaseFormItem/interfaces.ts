// Interfaces
import { FormItemProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseFormItemProps extends FormItemProps {
  labelColor?: APP_COLOR | APP_COLOR_LIGHT
  labelSize?: number
}
