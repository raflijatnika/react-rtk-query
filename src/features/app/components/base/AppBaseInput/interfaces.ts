// Interfaces
import { InputProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseInputProps extends InputProps {
  backgroundColor?: APP_COLOR | APP_COLOR_LIGHT
  height?: number
  width?: number
}
