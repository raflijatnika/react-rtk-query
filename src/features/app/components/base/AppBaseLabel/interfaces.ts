// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseLabelProps {
  fontWeight?: number
  fontSize?: number
  isBold?: boolean
  fontColor?: APP_COLOR | APP_COLOR_LIGHT
}
