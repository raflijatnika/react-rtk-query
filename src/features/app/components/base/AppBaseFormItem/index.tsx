// Styled Components
import styled from 'styled-components'

// Antd
import { Form } from 'antd'

// Constant
import { APP_COLOR } from '@/features/app/constant/app-style.constant'

// Interfaces
import { IAppBaseFormItemProps } from './interfaces'

const AppBaseFormItem = styled(Form.Item).attrs(
  (props): IAppBaseFormItemProps => props
)<IAppBaseFormItemProps>`
  .ant-form-item-label label {
    color: ${props => props?.labelColor || APP_COLOR.DARK};
    font-size: ${props => (props?.labelSize ? `${props.labelSize}px` : '16px')};
    line-height: 20.83px;
    font-weight: 500px;
    padding: 0px;
  }
`

export { AppBaseFormItem }
