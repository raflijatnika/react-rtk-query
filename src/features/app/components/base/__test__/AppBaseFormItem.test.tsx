// Test
import { render } from '@testing-library/react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'
import 'jest-styled-components'

// Components
import { AppBaseFormItem } from '../AppBaseFormItem'

// Interfaces
import { IAppBaseFormItemProps } from '../AppBaseFormItem/interfaces'

// Constant
import { APP_COLOR } from '@/features/app/constant/app-style.constant'

const rendererInstance = (props?: IAppBaseFormItemProps): ReactTestRenderer =>
  renderer.create(<AppBaseFormItem {...props} />)

const options: jest.Options = {
  modifier: '.ant-form-item-label label'
}

it('AppBaseFormItem should render', () => {
  render(<AppBaseFormItem />)
})

it('AppBaseFormItem should have snapshot', () => {
  const tree = rendererInstance().toJSON()

  expect(tree).toMatchSnapshot()
})

it('AppBaseFormItem should have default style', () => {
  const tree = rendererInstance().toJSON()

  expect(tree).toHaveStyleRule('line-height', '20.83px', options)
  expect(tree).toHaveStyleRule('font-weight', '500px', options)
  expect(tree).toHaveStyleRule('padding', '0px', options)
  expect(tree).toHaveStyleRule('color', APP_COLOR.DARK, options)
  expect(tree).toHaveStyleRule('font-size', '16px', options)
})

it('AppBaseFormItem should render correct props', () => {
  const tree = rendererInstance({
    labelColor: APP_COLOR.WHITE,
    labelSize: 20
  }).toJSON()

  expect(tree).toHaveStyleRule('color', APP_COLOR.WHITE, options)
  expect(tree).toHaveStyleRule('font-size', '20px', options)
})
