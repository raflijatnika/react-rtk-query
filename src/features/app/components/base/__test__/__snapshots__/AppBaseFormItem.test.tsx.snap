// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBaseFormItem should have snapshot 1`] = `
.c0 .ant-form-item-label label {
  color: #06152B;
  font-size: 16px;
  line-height: 20.83px;
  font-weight: 500px;
  padding: 0px;
}

<div
  className="ant-form-item c0"
>
  <div
    className="ant-row ant-form-item-row"
    style={Object {}}
    theme={Object {}}
  >
    <div
      className="ant-col ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        />
      </div>
    </div>
  </div>
</div>
`;
